app/models/hotel.rb

```diff
@@ -1,3 +1,12 @@
 class Hotel < ApplicationRecord
   has_many :rooms
-end
\ No newline at end of file
+
+
+# this is as far as I got. Couldn't figure out how to call all the guests linked to the hotels
+
+  def guests_at_hotel
+    Guest.find_each do |guest|
```

Comment:
There are 2 main ways you could do this:
1. Query with a join from rooms. From the hotel instance, grab the rooms associated with it, and join those rooms to guests (remember that you can do this directly -- `.joins(:guests)` -- because of the associations you've set up with has_many :guests via the joins table). Then select the name of the distinct guests. At the end of this comment is a possible query that would work but I didn't paste it up here in case you want to try this out first and then look at a solution. 

2. Another option here is adding additional associations! I like to call these mooching associations. Hotels have many rooms, and rooms have associations to guests. Rails allows us to create shortcuts with additional associations in order to make joins easier (but they would be doing the above query under the hood). In this Hotel model, you could add the association `has_many :guests, through: :rooms`. In this case, rooms is its child object rather than a joins object, but it still is the link that allows us to travel from the hotels table to the guests table via joins. Once you have this association, you can directly call .guests from a hotel. So that would allow you to just do `self.guests.distinct` from a hotel instance method. 

Look below for query answer
~~~~~~~~~~~~~~~~~
From Hotel instance method: self.rooms.joins(:guests).select("distinct guests.name as guest_name")
From this returned AR relation, you could iterate and call `guest_name` to grab the guest name.